clients/unshaded/src/main/java/tachyon/client/TachyonFS.java

```diff
@@ -642,23 +642,7 @@ public synchronized ClientFileInfo getFileStatus(int fileId, boolean useCachedMe
   public synchronized String getLocalBlockTemporaryPath(long blockId, long initialBytes)
       throws IOException {
     String blockPath = mWorkerClient.requestBlockLocation(blockId, initialBytes);
-
-    File localTempFolder;
-    try {
-      localTempFolder = new File(CommonUtils.getParent(blockPath));
-    } catch (InvalidPathException e) {
-      throw new IOException(e);
-    }
-
-    if (!localTempFolder.exists()) {
-      if (localTempFolder.mkdirs()) {
-        CommonUtils.changeLocalFileToFullPermission(localTempFolder.getAbsolutePath());
-        LOG.info("Folder {} was created!", localTempFolder);
-      } else {
-        throw new IOException("Failed to create folder " + localTempFolder);
-      }
-    }
-
+    CommonUtils.createBlockPath(blockPath);
```

Comment:
Perhaps this should be handled in the worker as well? Though does not have to be in this PR. Then createBlockPath can go into the (not yet merged) BlockWorkerUtils?